File: n2s.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-

from flask import Flask, request, render_template
import json
import sys
import urllib.request

app = Flask(__name__)

@app.route('/n2s', methods=['GET'])
def n2s_show():
    return render_template('n2s.html')

@app.route('/n2s', methods=['POST'])
def n2s():
	errRetrive = "No data retrived. \nPlease check the playlist id again."
	playlistId = request.form['playlistId']
	postContent = {"TransCode":"020111","OpenId":"123456789","Body":{"SongListId":playlistId}}
	encodedContent = json.dumps(postContent).encode('utf-8')

	reqUrl = "https://api.hibai.cn/api/index/index"

	req = urllib.request.Request(reqUrl)
	req.add_header('Content-Type', 'application/json; charset=utf-8')
	req.add_header('Accept', 'application/json')

	response = urllib.request.urlopen(req, encodedContent).read()
	data = json.loads(response)

	output = []

	if data["ErrCode"] != "OK":
		print(data["ErrCode"])
		print(errRetrive)
		print(help)

	body = data["Body"]
	playlistName = body["name"]
	tracks = body["songs"]

	for track in tracks:
		trackName = track["title"]
		artist = track["author"]
		output.append(trackName + " - " + artist)  

	return render_template('n2s.html', output=output)

if __name__ == '__main__':
    app.run(host='0.0.0.0', port=8888, debug=True)

File: n2s.js
<!-- search and show -->
<script type="text/javascript">
$("#submit").click(function(){
var playlistId = $("#playlistId").val();
if(!playlistId){
playlistId = 9699105;
}
var url = "https://api.hibai.cn/api/index/index";
var postContent = {"TransCode":"020111","OpenId":"123456789","Body":{"SongListId":"9699105"}};
postContent.Body.SongListId=playlistId.toString();
$.ajax({
type: "POST",
url: url,
dataType: 'json',
data: postContent,
success: function(result){
console.log(result); // 'name', 'age', 'city'
var list = [];
var body = result.Body;
var playlistName = body.name;
var tracks = body.songs;
console.log(body);
console.log(playlistId);
console.log(tracks);
console.log(tracks.length);

for (i=0; i<tracks.length; i++){
    var title = tracks[i].title;
    var author = tracks[i].author;
    var custom = title +' - '+ author +'\n' ;
    list.push(custom);
    }

    console.log(list);
    console.log(list);
    list = list.join("");
    $("#output").val(list);
    $('#plname').text(playlistName);
    $('#numsongs').text(tracks.length);
    return false;
    },
    error: function(request) {
    alert("Connection error");
    return false;
    }
    });
    }
    );
    </script>
    <!-- copy -->
    <script>
    function copyArticle(event){
    const range = document.createRange();
    range.selectNode(document.getElementById('output'));
    const selection = window.getSelection();
    if(selection.rangeCount > 0) selection.removeAllRanges();
    selection.addRange(range);

    document.execCommand('copy');
    }
    document.getElementById('copy').addEventListener('click', copyArticle, false);
    </script>
